scripts/setup-lxd.sh

```diff
@@ -1,12 +1,27 @@
-#!/bin/sh
+#!/bin/bash
 # Copyright 2016 Canonical Ltd.
 # Licensed under the AGPLv3, see LICENCE file for details.
-set -ex
+set -eux
 
 # Do the manual steps a user has to run on a fresh system to get an lxd
 # bridge so the juju lxd provider can function. Taken from changes made
 # to cloud-init to do approximately this.
 
+VERSION=$(lxd --version)
+
+# LXD 2.3+ needs lxdbr0 setup via lxc.
+if [[ "$VERSION" > "2.2" ]]; then
+    if [[ ! $(lxc network list | grep lxdbr0) ]]; then
+        # Configure a known address ranges for lxdbr0.
+        lxc network create lxdbr0 \
+            ipv4.address=10.0.8.1/24 ipv4.nat=true \
+            ipv6.address=d42:964b:da9f:f1a5::1/64 ipv6.nat=true
```

Comment:
Yes it is new. Maybe I should set ipv6.address=none ipv6.nat=false. We do need to start testing juju on ipv6 and saw this as an opportunity. This is probably wrong because the test will need to setup single or mixed networks to ensure the conditions are correct.